# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/tests/unit/test_x509.js
# Commit: 16350e08d34f
# Full Hash: 16350e08d34ff06c7f46745c6b160c7a7bda9140
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2019-12-06 04:43:57
# Regressor Bug: 1586855
# File Overlap Count: 6
# Description:
#   bug 1586855 - incorporate CRLite filters into cert_storage r=jcj,kjacobs
#   
#   This patch implements CRLite lookups for TLS server certificate revocation
#   information in telemetry-only mode. It adds a new preference
#   "security.pki.crlite_mode" to control the behavior of this feature. Setting
# ==============================================================================

diff -r 64fd40663930 -r 16350e08d34f security/manager/ssl/tests/unit/test_x509.js
--- a/security/manager/ssl/tests/unit/test_x509.js	Thu Dec 05 22:23:54 2019 +0000
+++ b/security/manager/ssl/tests/unit/test_x509.js	Thu Dec 05 22:41:53 2019 +0000
@@ -5,9 +5,7 @@
 
 // Tests X509.jsm functionality.
 
-var { X509 } = ChromeUtils.import("resource://gre/modules/psm/X509.jsm");
-
-function stringToBytes(s) {
+function stringToArray(s) {
   let b = [];
   for (let i = 0; i < s.length; i++) {
     b.push(s.charCodeAt(i));
@@ -16,7 +14,7 @@
 }
 
 function readPEMToBytes(filename) {
-  return stringToBytes(atob(pemToBase64(readFile(do_get_file(filename)))));
+  return stringToArray(atob(pemToBase64(readFile(do_get_file(filename)))));
 }
 
 function run_test() {
@@ -82,7 +80,7 @@
   );
   deepEqual(
     certificate.tbsCertificate.issuer.rdns[0].avas[0].value.value,
-    stringToBytes("Test CA"),
+    stringToArray("Test CA"),
     "expired-ee.pem should have issuer 'Test CA'"
   );
 
@@ -109,7 +107,7 @@
   );
   deepEqual(
     certificate.tbsCertificate.subject.rdns[0].avas[0].value.value,
-    stringToBytes("Expired Test End-entity"),
+    stringToArray("Expired Test End-entity"),
     "expired-ee.pem should have subject 'Expired Test End-entity'"
   );
 